shoop/front/apps/customer_information/views.py

```diff
@@ -57,11 +36,49 @@ def form_valid(self, form):
         if shipping_address.pk != contact.default_shipping_address_id:  # Identity changed due to immutability
             contact.default_shipping_address = shipping_address
 
-        user.email = contact.email
-        user.first_name = contact.first_name
-        user.last_name = contact.last_name
-        user.save()
+        if not bool(get_company_contact(self.request.user)):  # Only update user details for non-company members
```

Comment:
Because then the username would switch every time the user edited either the person contact or the company contact, which I don't think was the intended behavior. We might be missing some kind of indication to the user which email would be linked to their account (in this case, the company email).